### ISSUE TITLE
In Filesystem.php line 315: symlink(): No such file or directory

### ISSUE THREAD

**Issue by skliarie** [ammannbe/RecipeManager#12]
##### Bug or feature request?

-   [x] bug
-   [ ] feature request

##### Already searched?

-   [x] I have already searched in the issues

##### Describe the bug or feature

Started the docker stack:
`root@t11:/opt/recipe-manager# docker-compose up`
And keep getting this error on stdout:
recipe-manager_app_1 exited with code 1
app_1          | 
app_1          | In Filesystem.php line 315:
app_1          |                                         
app_1          |   symlink(): No such file or directory  
app_1          |                                         
app_1          |

##### Reproduce (if bug)

1. Follow https://github.com/ammannbe/RecipeManager/tree/7.x/storage/docker
2. Run docker-compose up

##### Desktop (if bug)

-   OS: ubuntu 20.04.02
```
root@t11:/opt/recipe-manager# docker-compose -v
docker-compose version 1.25.0, build unknown
root@t11:/opt/recipe-manager# docker -v
Docker version 20.10.7, build f0df350
root@t11:/opt/recipe-manager# dpkg -l | grep docker
ii  docker-ce                                  5:20.10.7~3-0~ubuntu-focal
ii  docker-compose                             1.25.0-1
...
```

**Comment by ammannbe** [comment]
Hi @skliarie 
Thanks for your report.

This error comes from the [entrypoint.sh](https://github.com/ammannbe/RecipeManager/blob/2d1be5698224b3e7afc8aa9f383433c1566bcf83/storage/docker/root/usr/local/bin/entrypoint.sh#L13)
The command `php artisan storage:link` should create a symlink from `/var/www/html/storage/app` to `/var/www/html/public/storage` (in your docker container)

Please try to delete this symlink in your named volume and start your container again.
Either with something like: `rm /var/lib/docker/volumes/recipe-manager_html/_data/public/storage`
or when the container is running with: `docker-compose down -v` (the -v stands for removing named volumes, as the `html` volume is one).

You can debug these files/folders with replacing the named volume `html` in your docker-compose.yml with e.g. `./html:/var/www/html`

Let me know if that helped

**Comment by skliarie** [comment]
I created html and replaced named volume html in my docker-compose.yml :
./html:/var/www/html

The html directory became populated:
```
root@t11:/opt/recipe-manager# ls -la html/
total 12
drwxr-xr-x 3 root root 4096 Jun 21 22:14 .
drwxr-xr-x 4 root root 4096 Jun 21 22:14 ..
-rwxr-xr-x 1 root root    0 Jun 21 22:14 .env
drwxr-xr-x 3 root root 4096 Jun 21 22:14 storage
root@t11:/opt/recipe-manager# du html
4       html/storage/app
8       html/storage
12      html
```

There is a different problem now:
```
app_1          | Could not open input file: artisan
recipe-manager_app_1 exited with code 1
```
And then the recipe-manager_app_1 was continuously restarting..
recipe-manager_app_1           suhype/recipe-manager:7.0      7b22437ac48d   Restarting (1) 18 seconds ago

**Comment by ammannbe** [comment]
Sorry my fault... When you mount a host directory into container, the contents of the host directory shadows the contents of the container. So my second suggestion won't work that easy.

I just tried to reproduce it and found the problem.
If the symlink `./data/app/public` exists or the directory `./data/app/public` is missing, you can run following commands:
```bash
# Delete the symlink
rm ./data/app/public
# Create the required directory
mkdir ./data/app/public
# Change the ownership, in my case User: 1000 ; Group: 1000
chown 1000:1000 ./data/app/public
```

After that run the container again and it should work.

**Comment by skliarie** [comment]
I did the suggested:
```

root@t11:/opt/recipe-manager# ls -lnd ./data/app
drwxr-xr-x 4 1000 1000 4096 Jun 22 10:17 ./data/app
root@t11:/opt/recipe-manager# ls -lnd ./data/app/public
drwxr-xr-x 2 1000 1000 4096 Jun 22 10:17 ./data/app/public

```
And get same error:
```
recipe-manager_app_1 exited with code 1
app_1          | Could not open input file: artisan

```

**Comment by ammannbe** [comment]
Please revert these changes first
> You can debug these files/folders with 
replacing the named volume html in your docker-compose.yml with e.g. ./html:/var/www/html

**Comment by skliarie** [comment]
Reverted the changes. The html is named volume now (instead of host directory).
Started, stopped, replaced the symlink:
```
root@t11:/opt/recipe-manager# ls -ldn data/app/public
drwxr-xr-x 2 1000 1000 4096 Jun 22 11:05 data/app/public

```
started again, after some time got different error (posting here, let me know if I should open separate ticket):
```
app_1          | In Connection.php line 678: 
app_1          |                                                                                
app_1          |   SQLSTATE[42S02]: Base table or view not found: 1146 Table 'recipe_manager.r  
app_1          |   ecipes' doesn't exist (SQL: select * from `recipes` where `recipes`.`delete  
app_1          |   d_at` is null and ((`user_id` is null) or (`cookbook_id` is null)) order by  
app_1          |    `id` asc limit 500)                                                         
app_1          |                                                                                 
app_1          | 
app_1          | In Connection.php line 338:
app_1          |                                                                                
app_1          |   SQLSTATE[42S02]: Base table or view not found: 1146 Table 'recipe_manager.r  
app_1          |   ecipes' doesn't exist   
```

**Comment by ammannbe** [comment]
That means that either your database is still not up and running or you have a broken database.
You can try to recreate your mysql/mariadb docker container and delete the database (the files should be in `./data/mysql`)

Note: the first run of your sql container needs some more time to create the database structure etc. So it can happen that the application is running, but the sql server is not ready. But after the second run it should work fine.

**Comment by skliarie** [comment]
I tried to run the stack several times, but the db was not populated.
This is the app container output:
```
app_1          | The [/var/www/html/public/storage] link already exists.
app_1          | The links have been created.
app_1          | **************************************
app_1          | *     Application In Production!     *
app_1          | **************************************
app_1          | 
app_1          | Command Canceled!
app_1          | Configuration cache cleared!
app_1          | Configuration cached successfully!
app_1          | Route cache cleared!
app_1          | Routes cached successfully!
app_1          | Compiled views cleared!
app_1          | Blade templates cached successfully!
app_1          | 
app_1          | > prod
app_1          | > npm run production
app_1          | 
app_1          | 
app_1          | > production
app_1          | > cross-env NODE_ENV=production node_modules/webpack/bin/webpack.js --no-progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js
app_1          |  DONE  Compiled successfully in 35614ms11:59:50 AM
```
The db has no tables defined:
```
root@t11:/opt/recipe-manager# ls -ln data/mysql/recipe_manager/
total 4
-rw-rw---- 1 999 999 67 Jun 22 14:56 db.opt

```
Could be the "wrong" data/app/public link on first run prevents the db to be properly populated? And after I "fix" it - the population is not retried?

**Comment by skliarie** [comment]
The db got populated only after I ran "php artisan migrate:install" from the app container.
After that I was able to see the website in browser.
Nothing worked though. Any submit or click generated request to http://localhost:8000/WHATEVER
For example http://localhost:8000/api/auth/register

**Comment by skliarie** [comment]
The db got populated only after I ran "php artisan migrate:install" from the app container.
Added domain to .env, stuff seems to work now, thank you!
